tensorflow/python/ops/script_ops.py

```diff
@@ -171,8 +160,48 @@ def _convert(value, dtype=None):
       return np.asarray(value, order="C", dtype=object)
     elif result.dtype.char == "U":
       return result.astype(np.bytes_)
-    else:
-      return result
+    return result
+
+  def __call__(self, args):
+    ret = self._func(*args)
+    # Strings seem to lead to a memory leak here if they're not wrapped in a
+    # list.
+    if isinstance(ret, six.binary_type):
+      ret = [ret]
+    # Ensures that we return either a single numpy array or a list of numpy
+    # arrays.
+    if isinstance(ret, (tuple, list)):
+      return [
+          self._convert(x, dtype=dtype.as_numpy_dtype)
+          for (x, dtype) in zip(ret, self._out_dtypes)
+      ]
+    return self._convert(ret)  # TODO: pass self._out_dtypes?
```

Comment:
Is there any reason to not pass out_dtypes here?

Reply:
One thing - the registry will force the Tout to be an `array / list` -- see ~line 270 in the `_internal_py_func` where the `Tout` is initialized.  So we would need to sanitize the _out_dtypes. But that should be easy.